### PR TITLE
Issue / 404 Page

### PR DESCRIPTION
**Pull request by SeferMirza**
Learn how to use 404 page

## Tasks

- [x] learn how to use 404 page
  - In ContentDoc and ContentQuery components, we can display the template 'page not found' error from `#not-found` tag instead of content
  - when querycontent returns null, we can trigger error page with createError.
  - [x] Error page on Nuxt
  - [x] Error page content on Nuxt Content Module
    - If we assign query in `#not-found` and it fails, `error.vue` goes but it doesn't show full page. it comes after refreshed.
    - When query is written in `#not-found`, the query is not thrown every time the page is loaded. If the content is not found, the `#not-found` is executed and the query query is thrown after that.
  - nuxt content and nuxt's `error.vue` are not compatible. When you get a full page error, you need to clear the errors with `clearError`.
- [x] add example usage
  - [x] Nuxt error page
    - [x] add error.vue
    - [x] test trigger error page with `queryContent()`
    - [x] test that error.vue is shown when route does not access any .vue file under pages folder.
      > Since there is a direct [page] under pages, it catches every invalid route with nuxt content. When I deleted this, error.vue was triggered
  - [x] Nuxt content error page
    - [x] test error content with `<ContentQuery>` and `<ContentDoc>`
    - [x] add error.md and show when page not found
- [x] add document

ℹ️ error page was removed because it breaks generate and it doesn't make sense to try to create a page that doesn't exist. I mentioned in the document why we use it if we use it.

## Additional Task

- [x] fix generate
- [x] eslint keyword-spacing rule https://eslint.org/docs/latest/rules/keyword-spacing#overrides
- [x] add workflow for every push in pr (generate test)

### CODE REVIEW

File: .theme/pages/demo/error-handling/trigger-error-page.vue [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+</template>
+<script setup>
+import { showError } from "#app";
+import { queryContent } from "#imports";
+
+let index;
+
+// Searching for a non-existent markdown
+try{
+  index = await queryContent("/demo/non-exist-mardown")
```
Comment (cihandeniz) [inline .theme/pages/demo/error-handling/trigger-error-page.vue:14]: non-exist -> non-existent

File: .theme/pages/demo/error-handling/trigger-error-page.vue [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  </div>
+</template>
+<script setup>
+import { showError } from "#app";
+import { queryContent } from "#imports";
+
+let index;
+
+// Searching for a non-existent markdown
+try{
```
Comment (cihandeniz) [inline .theme/pages/demo/error-handling/trigger-error-page.vue:13]: `try{` -> `try {`

bu suslu parantez oncesine bosluk eslint'te yok sanirim, onu ekleyebilirsen super olur

File: .theme/pages/demo/error-handling/content-component.vue [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div>
+    <ContentDoc>
+      <template #not-found>
+        <ContentQuery
+          v-slot="{ data }"
+          path="/"
+          :where="{ _path: { $eq: '/error' }}"
```
Comment (cihandeniz) [inline .theme/pages/demo/error-handling/content-component.vue:8]: `path="/error"` yaparsan `:where` tarafina gerek kalmaz diye tahmin ediyorum

**Review by cihandeniz — COMMENTED** [review]
henuz bitmedi ama gordugum minor kozmetik seyleri soyleyeyim dedim, bunlari toparlarsan pair yaptigimizda direkt tasarimi konusuruz

File: .theme/pages/demo/error-handling/trigger-error-page.vue (hunk 1 above)
Comment (cihandeniz) [reply .theme/pages/demo/error-handling/trigger-error-page.vue]: trigger error page dosyasi kalkmis ama sen hallettim anlamina gelen ikonu koymussun, kafa karistiriyorsun, biraz empati yapalim

File: .theme/pages/demo/error-handling/trigger-error-page.vue (hunk 2 above)
Comment (cihandeniz) [reply .theme/pages/demo/error-handling/trigger-error-page.vue]: bunu da bulamadim, ama bir sey warn olmus, ve if'lerden sonra space gelmis alakasiz. 

duz parantez'den once space istemedim `if(`, susluden once parantez dedim `try {`.

genel olarak isin bitmedigini biliyorum, ama bu comment'e sapka yollamissin ama anlamadim neyi cozdugunu

Comment (SeferMirza) [reply .theme/pages/demo/error-handling/trigger-error-page.vue]: eslint te rule ları arattığım zaman bunun kuralı için keyword-spacing i buldum. bu kuralı açınca try dan sonraki '{' ile if ten sonraki '(' i aynı kuralı uyguluyor. bunları ayırmak için override opsiyonu koymuşlar son son commit ile onu  ekledim şimdi artık 'if(' şeklinde 'try {' şeklinde istiyor zorunlu olarak

**Review by cihandeniz — APPROVED** [review]

**Pull request merged**